File: day4.php
<?php
$vnos = fopen("day4.txt", "r") or die("Unable to open file!");
$podatki = fread($vnos,filesize("day4.txt"));
fclose($vnos);
$razbito = preg_split("/\n/", $podatki);
$skupaj = 0;
for($i = 0; $i < count($razbito); $i++) {
	for($j=0;$j < strlen($razbito[$i]); $j++) {
			if($razbito[$i][$j] == "X") {
				$prostorLevo = $j > 2;
				$prostorDesno = $j < strlen($razbito[$i]) -3;
				$prostorGor = $i > 2;
				$prostorDol = $i < count($razbito) -3;
				if($prostorLevo) {
					if($razbito[$i][$j-1] == "M" && $razbito[$i][$j-2] == "A" && $razbito[$i][$j-3] == "S"){
						$skupaj++;
					}

					//diagonala gor
					if($prostorGor) {
						if($razbito[$i-1][$j-1] == "M" && $razbito[$i-2][$j-2] == "A" && $razbito[$i-3][$j-3] == "S"){
							$skupaj++;
						}
					}
					//diagonala dol
					if($prostorDol) {
						if($razbito[$i+1][$j-1] == "M" && $razbito[$i+2][$j-2] == "A" && $razbito[$i+3][$j-3] == "S"){
							$skupaj++;
						}
					}
				}
				//desno
				if($prostorDesno) {
					if($razbito[$i][$j+1] == "M" && $razbito[$i][$j+2] == "A" && $razbito[$i][$j+3] == "S"){
						$skupaj++;
					}
					//diagonala gor
					if($prostorGor) {
						if($razbito[$i-1][$j+1] == "M" && $razbito[$i-2][$j+2] == "A" && $razbito[$i-3][$j+3] == "S"){
							$skupaj++;
						}
					}
					//diagonala dol
					if($prostorDol) {
						if($razbito[$i+1][$j+1] == "M" && $razbito[$i+2][$j+2] == "A" && $razbito[$i+3][$j+3] == "S"){
							$skupaj++;
						}
					}
				}

				if($prostorDol) {
					if($razbito[$i+1][$j] == "M" && $razbito[$i+2][$j] == "A" && $razbito[$i+3][$j] == "S"){
						$skupaj++;
					}
				}
				if($prostorGor) {
					if($razbito[$i-1][$j] == "M" && $razbito[$i-2][$j] == "A" && $razbito[$i-3][$j] == "S"){
						$skupaj++;
					}
				}
			}
	}
}
echo 'part 1: '.$skupaj;

File: day4.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

namespace day4
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string input = @"day4.txt";
            if (File.Exists(input))
            {
                string readInput = File.ReadAllText(input);
                string[] vrstica = Regex.Split(readInput, "\n");
                int skupaj = 0;
                for(var i = 0; i < vrstica.Length; i++)
                {
                    for(var j = 0; j < vrstica[i].Length; j++)
                    {
                        if (vrstica[i][j] == 'X')
                        {
                            var prostorDesno = j < (vrstica[i].Length - 3);
                            var prostorLevo = j > 2;
                            var prostorGor = i > 2;
                            var prostorDol = i < (vrstica.Length - 3);
                            //LEVO
                            if(prostorLevo)
                            {
                                if (vrstica[i][j-1] == 'M' && vrstica[i][j-2] == 'A' && vrstica[i][j-3] == 'S')
                                {
                                    skupaj++;
                                }
                                //DIGONALA GOR
                                if(prostorGor)
                                {
                                    if (vrstica[i - 1][j-1] == 'M' && vrstica[i - 2][j-2] == 'A' && vrstica[i-3][j-3] == 'S')
                                    {
                                        skupaj++;
                                    }
                                }
                                //DIAGONALA DOL
                                if(prostorDol)
                                {
                                    if (vrstica[i +1][j-1] == 'M' && vrstica[i+2] [j-2] == 'A' && vrstica[i+3][j-3] == 'S')
                                    {
                                        skupaj++;
                                    }
                                }
                            }
                            //DESNO
                            if(prostorDesno)
                            {
                                if (vrstica[i][j+1] =='M' && vrstica[i][j+2] == 'A' && vrstica[i][j+3] == 'S')
                                {
                                    skupaj++;
                                }
                                //DIAGONALA GOR
                                if (prostorGor)
                                {
                                    if (vrstica[i - 1][j + 1] == 'M' && vrstica[i - 2][j + 2] == 'A' && vrstica[i - 3][j + 3] == 'S')
                                    {
                                        skupaj++;
                                    }
                                }
                                //DIAGONALA DOL
                                if(prostorDol)
                                {
                                    if (vrstica[i + 1][j+1] =='M' && vrstica[i + 2][j+2] =='A' && vrstica[i + 3][j+3] == 'S')
                                    {
                                        skupaj++;
                                    }
                                }
                            }
                            //DOL
                            if(prostorDol)
                            {
                                if (vrstica[i + 1][j] == 'M' && vrstica[i + 2][j] == 'A' && vrstica[i + 3][j] == 'S')
                                {
                                    skupaj++;
                                }
                            }
                            //GOR
                            if(prostorGor)
                            {
                                if (vrstica[i - 1][j] == 'M' && vrstica[i - 2][j] == 'A' && vrstica[i-3][j] == 'S')
                                {
                                    skupaj++;
                                }
                            }
                        }
                    }
                }
                Console.WriteLine("part 1:" + skupaj);
            }
        }
    }
}
